src/js/components/pages/scene-editor-gui.jsx

```diff
@@ -0,0 +1,793 @@
+var React = require('react');
+var _ = require('lodash');
+var SceneStore = require('../../stores/scene-store');
+var GridStore = require("../../stores/grid-store");
+var HubSendActions = require('../../actions/hub-send-actions');
+var SceneActions = require('../../actions/scene-actions');
+var Rnd = require('react-rnd');
+var FontAwesome = require('react-fontawesome');
+var Rectangle = require('react-rectangle');
+var ImageLoader = require('react-imageloader');
+
+var saveTimeout = null;
+
+var SceneEditorGUI = React.createClass({
+
+    getInitialState: function () {
+        return _.extend(this.getStateFromStores(), {
+            focusedMediaObject: null,
+            scene: null,
+            placement: null,
+            mediaObject: null,
+            mediaTypeSupported: null,
+            shouldSave: false,
+            height: 0,
+            width: 0,
+        });
+    },
+
+    getStateFromStores: function () {
+        return {
+            scene: SceneStore.getScene(this.props._id),
+        };
+    },
+
+    componentDidMount: function () {
+        console.log("SceneEditorGUI: Mounted")
+
+        //subscribe to changes
+        SceneStore.addChangeListener(this._onChange);
+        GridStore.addChangeListener(this._onFocusChange)
+
+        //this is the first point where rendered size of area can be captured
+        this.setState({width: this.SceneLayoutArea.offsetWidth, height: this.SceneLayoutArea.offsetHeight})
+    },
+
+    componentWillReceiveProps: function (nextProps) {
```

Comment:
This lifecycle function can be used to get the resizing of the component in gear based on the updated size of the parent.